microsoft/Windows.UI.Composition-Win32-Samples
# Update C++ samples to use Visual Studio 2022, Platform Toolset v143, and the latest C++/WinRT

This change updates all C++ samples to build properly using the latest build tools (VS 2022, and v143 of the platform toolset) as well as the latest version of C++/WinRT. Specifically:

  * Target Platform Toolset v143 in all C++ projects
  * Consume the latest C++/WinRT nuget package instead of using the SDK provided projections
  * Target the "latest" Windows SDK
  * Update READMEs for projects updated to use Visual Studio 2022
  * Remove `WINRT_WARNING_PUSH` usage (no longer exists in newer versions of C++/WinRT)
  * Small tweaks to get projects building again

There is also a bonus change to the VisualLayerIntegration sample to prevent dereferencing null.
